src/components/popup/tx-item.tsx

```diff
@@ -126,6 +101,120 @@ const SpeedUpButton = ({
+      ))}
+    </>
+  );
+};
+
+const TxItemRow: React.FC<{
+  title: string;
+  value: number | string | null;
+  caption?: string;
+  icon: JSX.Element;
+  onClick?: () => Window | null;
+}> = ({ title, caption, value, icon, onClick, ...rest }) => {
+  const [component, bind] = usePressable(true);
+  return (
+    <Box position="relative" cursor="pointer" {...bind} {...rest}>
+      <Stack
+        alignItems="center"
+        spacing="base-loose"
+        isInline
+        position="relative"
+        zIndex={2}
+        onClick={onClick}
+      >
+        {icon}
+        <SpaceBetween flexGrow={1}>
+          <Stack spacing="base-tight">
+            <Title as="h3" fontWeight="normal">
+              {title}
+            </Title>
+            <Stack isInline flexWrap="wrap">
+              <Caption variant="c2">{caption}</Caption>
+            </Stack>
+          </Stack>
+          <Stack alignItems="flex-end" spacing="base-tight">
+            {value && (
+              <Title as="h3" fontWeight="normal">
+                {value}
+              </Title>
+            )}
+          </Stack>
+        </SpaceBetween>
+      </Stack>
+      {component}
+    </Box>
+  );
+};
+
+const StxTransferItem: React.FC<{
+  stxTransfer: StxTransfer;
+  parentTx: AddressTransactionWithTransfers;
+}> = ({ stxTransfer, parentTx }) => {
+  const currentAccount = useCurrentAccount();
+  const { handleOpenTxLink } = useExplorerLink();
+  const title = 'Stacks Token Transfer';
+  const caption = getTxCaption(parentTx.tx) ?? '';
+  const isOriginator = stxTransfer.sender === currentAccount?.address;
+
+  const value = `${isOriginator ? '-' : ''}${stacksValue({
+    value: stxTransfer.amount,
+    withTicker: false,
+  })}`;
+
+  const icon = isOriginator ? IconArrowUp : IconArrowDown;
+
+  const iconWrapper = (
+    <Circle position="relative" size="36px" bg={color('accent')} color={color('bg')}>
+      <Box as={StxIcon} />
+      <TypeIconWrapper icon={icon} bg={'brand'} />
+    </Circle>
+  );
+
+  return (
+    <TxItemRow
+      title={title}
+      caption={caption}
+      value={value}
+      onClick={() => handleOpenTxLink(parentTx.tx.tx_id)}
+      icon={iconWrapper}
+    />
+  );
+};
+
+export const TxWithTransferItem: React.FC<TxWithTransferItemProps & BoxProps> = ({
+  transaction,
+  ...rest
+}) => {
+  if (!transaction) return null;
+
+  if (!isAddressTransactionWithTransfer(transaction))
+    return <TxItem transaction={transaction as Tx} {...rest} />; // This is a normal Transaction or MempoolTransaction
+
+  const transact = transaction as AddressTransactionWithTransfers;
+
+  // Show transfer only for contract calls
+  if (transact.tx.tx_type !== 'contract_call')
+    return <TxItem transaction={transact.tx} {...rest} />;
+
+  return (
+    <>
+      <TxTransfers transaction={transact} />{' '}
+      {/* Show all the other transfers (made with a contract_call) */}
+      <TxItem transaction={transact.tx} {...rest} />
```

Comment:
These returns are all a bit confusing and the code comments aren't helping me understand what is happening, can you clarify? It seems to be returning the same `<TxItem />`?

Reply:
I moved the comment above.
TxTransfers are just the internal stx transfers displayed above the 'real' transaction. TxTransfers does not return TxItem  but a new kind of activity <StxTransferItem/>

Reply:
Actually the prettier moved the position of my comments 😞